### PR TITLE
fix: Fix Fatras end of world

### PR DESCRIPTION
**Pull request by andiwand**
Sometimes the Fatras propagation will not terminate and step with `double max` distance which results in FPEs

After these changes the FPEs during propagation disappeared

### CODE REVIEW

**Review by paulgessinger — APPROVED** [review]
So this introduces both an end of world check in the simulation actor, and also sets up and wires through the regular maximum step size and path length configuration options, correct?

**Comment by andiwand** [comment]
> So this introduces both an end of world check in the simulation actor, and also sets up and wires through the regular maximum step size and path length configuration options, correct?

yes exactly - the end of the world check and path limit should terminate the propagation if we are going too far and the step size limit should protect us from an unconstrained straight line step into infinity

File: Examples/Algorithms/Fatras/include/ActsExamples/Fatras/FatrasSimulation.hpp [hunk 1]
```diff
@@ -78,6 +78,11 @@ class FatrasSimulation final : public IAlgorithm {
     /// have associated material.
     bool generateHitsOnPassive = false;
 
+    /// Absolute maximum step size
+    double maxStepSize = 1 * Acts::UnitConstants::m;
+    /// Absolute maximum path length
+    double pathLimit = 10 * Acts::UnitConstants::m;
```
Comment (andiwand) [inline Examples/Algorithms/Fatras/include/ActsExamples/Fatras/FatrasSimulation.hpp:84]:
```suggestion
    double pathLimit = 30 * Acts::UnitConstants::m;
```

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [1250cfc132cb7ef36491286faefa5f5dc6045344]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426771) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426770) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426769) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426768) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426767) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426766) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426765) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426764) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426763) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426762) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426761) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426760) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426759) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426758) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426757) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33426756) |

**Review by paulgessinger — COMMENTED** [review]
This picked up conflicts now.

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**